### ISSUE TITLE
bug(ingest): event parser rejects numeric timestamps — real JS/Python Sentry SDK events are silently dropped

### ISSUE THREAD

**Issue by ajianaz** [codecoradev/trapfall#309]
## What
The envelope **event** parser requires `timestamp` as an RFC3339 **string**. Official Sentry SDKs (JavaScript, Python) send epoch **numbers** (e.g. `"timestamp": 1756278600.5`). Such events are dropped with only a server-side warning — the SDK receives 200 OK, so the loss is invisible to users.

Meanwhile the **transaction** parser is the opposite: it requires `timestamp` as `f64` and **rejects** RFC3339 strings.

## Evidence (2026-08-27, manual envelopes against develop @ f932d7d)
```
WARN dropping event: deserialize failed item="event"
     error=invalid type: integer `1756278600`, expected a string
WARN dropping transaction: deserialize failed item="transaction"
     error=invalid type: string "2026-08-27T07:52:30Z", expected f64
```

## Why it matters
- This is exactly the class of failure #299 (Sentry SDK compatibility validation) is meant to catch. Earlier integration testing (#251) likely passed because the Rust sentry crate serializes timestamps as strings.
- Silent event loss = the product fails at its one job.

## Suggested fix
Custom deserializer (or untagged enum) accepting both epoch numbers (secs, incl. float) and RFC3339 strings, normalized at parse time — for both events and transactions. Add regression tests with payloads captured from real @sentry/node and sentry-sdk Python.

**Comment by ajianaz** [comment]
New evidence from the gatehouse integration (codecoradev/gatehouse#70, merged 2026-09-22, `@sentry/bun` bridge → trapfall.ajianaz.dev): the merged bridge E2E test builds its event payload **without** a `timestamp` field, so this gap was never exercised by the integration validation. Any real `captureException` from a JS-family SDK carries an epoch-numeric `timestamp` — meaning real production events will be silently dropped (200 OK) even with the auth/DSN bridges in place. Requesting re-triage: this is the blocking interop bug for the gatehouse→trapfall integration, ahead of #327/#328.
